third_party/android-gmaps-addons/src/com/cocoahero/android/gmaps/addons/mapbox/MapBoxOfflineTileProvider.java

```diff
@@ -0,0 +1,209 @@
+package com.cocoahero.android.gmaps.addons.mapbox;
+
+import java.io.Closeable;
+import java.io.File;
+
+import android.database.Cursor;
+import android.database.sqlite.SQLiteDatabase;
+
+import com.google.android.gms.maps.model.LatLng;
+import com.google.android.gms.maps.model.LatLngBounds;
+import com.google.android.gms.maps.model.Tile;
+import com.google.android.gms.maps.model.TileProvider;
+
+public class MapBoxOfflineTileProvider implements TileProvider, Closeable {
```

Comment:
Can we avoid importing the whole third party source, and only include this class plus the LICENSE.md? Iirc it doesn't have any other dependencies.

Reply:
Yep, we can do that, I'll make the change.

Reply:
So, I've reduced the source as much as I possibly can in ebec4e4. Unfortunately, there are some files we need to make gradle work with the library, and I don't know enough about gradle to know if there's some other way to set this up to remove the need for things like the manifest file, etc. (it was already a bit of a pain to try to figure out this set up). 

At the very least, we now only have the two classes we need!